### DISCUSSION TITLE
Returning a structure value

### DISCUSSION THREAD

**Discussion by PrajwalBanakar26** [ezylang/EvalEx#373] [Q&A]
Version 3 documentation mentions that a structure with different data type values can be returned based on the expression evaluation. Could you please provide an example regarding the same?
[https://ezylang.github.io/EvalEx/concepts/changes.html#new-return-value-type](url)

**Comment by uklimaschewski** [comment]
Some examples for passing and using the different data types can be found [here](https://ezylang.github.io/EvalEx/concepts/datatypes.html).

Depending on your expression, different data types can be returned, e.g:
`a != b` will return an `EvaluationValue` of type `BOOLEAN`.
`"Hello " + "world"` will return an `EvaluationValue` of type `STRING`.
`1+2` will return an `EvaluationValue` of type `NUMBER`.

Structure and arrays could also be returned, in the standard set of functions there is currently only one function, that can do this.
The `IF(expression, returnValueIfTrue, returnValueIfFalse)` would return an array or structure, if parameters 2 and/or 3 are of this type.

But there might be also e.g. some array functions in the future, to sort, manipulate, calculate with arrays. These functions might return arrays again. Similar is possible for structures.
